tests/functional/simple_reference/test_simple_reference.py

```diff
@@ -0,0 +1,238 @@
+import pytest
+import os
+from tests.util import run_dbt, run_sql_file
+from tests.tables import TableComparison, get_tables_in_schema
+
+
+ephemeral_copy_sql = """
+{{
+  config(
+    materialized = "ephemeral"
+  )
+}}
+
+select * from {{ this.schema }}.seed
+"""
+
+ephemeral_summary_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select gender, count(*) as ct from {{ref('ephemeral_copy')}}
+group by gender
+order by gender asc
+"""
+
+incremental_copy_sql = """
+{{
+  config(
+    materialized = "incremental"
+  )
+}}
+
+select * from {{ this.schema }}.seed
+
+{% if is_incremental() %}
+
+    where id > (select max(id) from {{this}})
+
+{% endif %}
+"""
+
+incremental_summary_sql = """
+{{
+  config(
+    materialized = "table",
+  )
+}}
+
+select gender, count(*) as ct from {{ref('incremental_copy')}}
+group by gender
+order by gender asc
+"""
+
+materialized_copy_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select * from {{ this.schema }}.seed
+"""
+
+materialized_summary_sql = """
+{{
+  config(
+    materialized = "table"
+  )
+}}
+
+select gender, count(*) as ct from {{ref('materialized_copy')}}
+group by gender
+order by gender asc
+"""
+
+view_copy_sql = """
+{{
+  config(
+    materialized = "view"
+  )
+}}
+
+select * from {{ this.schema }}.seed
+"""
+
+view_summary_sql = """
+{{
+  config(
+    materialized = "view"
+  )
+}}
+
+select gender, count(*) as ct from {{ref('view_copy')}}
+group by gender
+order by gender asc
+"""
+
+view_using_ref_sql = """
+{{
+  config(
+    materialized = "view"
+  )
+}}
+
+select gender, count(*) as ct from {{ var('var_ref') }}
+group by gender
+order by gender asc
+"""
+
+
+@pytest.fixture
+def models():
+    return {
+        'ephemeral_copy.sql': ephemeral_copy_sql,
+        'ephemeral_summary.sql': ephemeral_summary_sql,
+        'incremental_copy.sql': incremental_copy_sql,
+        'incremental_summary.sql': incremental_summary_sql,
+        'materialized_copy.sql': materialized_copy_sql,
+        'materialized_summary.sql': materialized_summary_sql,
+        'view_copy.sql': view_copy_sql,
+        'view_summary.sql': view_summary_sql,
+        'view_using_ref.sql': view_using_ref_sql,
+    }
+
+
+@pytest.fixture
+def project_config_update():
+    return {
+        'vars': {
+            'test': {
+                'var_ref': '{{ ref("view_copy") }}',
```

Comment:
I don't see us using this anywhere below. I also wonder if we need to keep using dbt `--vars` for this kind of functionality? Could we instead update the pytest fixture (?) within the scope of a subsequent test? (I'm sure I'm misunderstanding fixture scope)

Reply:
This is used in the 'view_using_ref.sql' file. The 'project_config_update' fixture is merged into the basic project_config in the fixtures code. So this var is actually in the dbt_project.yml.